eventuate-crdt/src/main/scala/com/rbmhtechnology/eventuate/crdt/RGArray.scala

```diff
@@ -0,0 +1,165 @@
+  }
+
+  @tailrec
+  private def skipOver(pos: Position, idx: Int): Int = {
+    if (idx < vertices.length && vertices(idx).pos > pos) skipOver(pos, idx + 1) else idx
+  }
+}
+
+object RGArray {
+
+  def apply[A]: RGArray[A] =
+    new RGArray[A]()
+
+  implicit def RGArrayServiceOps[A] = new CRDTServiceOps[RGArray[A], Vector[A]] {
+    override def zero: RGArray[A] = RGArray.apply[A]
+
+    override def value(crdt: RGArray[A]): Vector[A] = crdt.value
+
+    override def prepare(crdt: RGArray[A], operation: Any): Try[Option[Any]] = operation match {
+      case InsertOp(pos, value, None) => Success {
+        Some(InsertOp(pos, value, Some(crdt.lastPos + 1)))
+      }
+      case op =>
+        super.prepare(crdt, op)
+    }
+
+    override def effect(crdt: RGArray[A], operation: Any, event: DurableEvent): RGArray[A] = operation match {
+      case InsertOp(after, value, Some(pos)) => crdt.insertRight(after, value.asInstanceOf[A], Position(pos, event.emitterId))
+      case DeleteOp(pos)                     => crdt.delete(pos, event.vectorTimestamp).prune(event.vectorTimestamp)
+    }
+  }
+}
+
+class RGArrayService[A](val serviceId: String, val log: ActorRef)(implicit val system: ActorSystem, val ops: CRDTServiceOps[RGArray[A], Vector[A]])
+  extends CRDTService[RGArray[A], Vector[A]] {
+
+  /**
+   * Adds `entry` to the RGA identified by `id` on the right side of the provided
+   * position and returns an updated entry set.
+   */
+  def insertRight(id: String, index: Position, value: A): Future[Vector[A]] = {
+    op(id, InsertOp(index, value))
+  }
+
+  /**
+   * Removes `entry` from the RGA identified by `id` and returns the updated entry set.
+   */
+  def delete(id: String, pos: Position): Future[Vector[A]] = op(id, DeleteOp(pos))
+
+  start()
+}
+
+case class PrepareInsert(after: Int, value: Any)
+case class PrepareDelete(at: Int)
+
+/**
+ * Persistent insert operation used for [[RGArray]].
+ */
+case class InsertOp(after: Position, value: Any, pos: Option[Int] = None) extends CRDTFormat
+
+/**
+ * Persistent delete operation used for [[RGArray]].
+ */
+case class DeleteOp(pos: Position) extends CRDTFormat
+
+/**
+ * Position of a particular element in relation to others within provided [[RGArray]].
+ * @param order Monotonically increasing order incremented with each insert.
+ * @param emitterId Identifier of a particular replica making an update.
+ */
+case class Position(order: Int, emitterId: String) extends Ordered[Position] {
+  override def compare(that: Position): Int = order compareTo that.order match {
+    case 0 => emitterId compareTo that.emitterId
+    case n => n
+  }
+}
+
+object Position {
+  val head: Position = Position(0, "")
+}
+
+/**
+ * Bucket for values stored inside an [[RGArray]].
+ * @param value Assigned value.
+ * @param pos Position of that value in relation to others within [[RGArray]].
+ * @param tombstone Determines a local casual time, at which current value has been removed.
```

Comment:
Rename to `deletionTime`? with documentation like *Determines the vector time at which this value has been removed*?